qst/src/main/java/io/deephaven/qst/table/MergeTable.java

```diff
@@ -0,0 +1,50 @@
+package io.deephaven.qst.table;
+
+import io.deephaven.qst.NodeStyle;
+import org.immutables.value.Value;
+import org.immutables.value.Value.Immutable;
+
+import java.util.List;
+
+@Immutable
+@NodeStyle
+public abstract class MergeTable extends TableBase {
+
+    public static MergeTable of(Table... tables) {
+        return MergeTable.builder().addTables(tables).build();
+    }
+
+    public static MergeTable of(Iterable<? extends Table> tables) {
+        return MergeTable.builder().addAllTables(tables).build();
+    }
+
+    public static Builder builder() {
+        return ImmutableMergeTable.builder();
+    }
+
+    public abstract List<Table> tables();
+
+    @Override
+    public final <V extends Visitor> V walk(V visitor) {
+        visitor.visit(this);
+        return visitor;
+    }
+
+    @Value.Check
+    final void checkSize() {
```

Comment:
Maybe the helper on QST table should just return the table if you try to merge it by itself.